src/github_runner_manager/reactive/consumer.py

```diff
@@ -100,12 +105,35 @@ def consume(
                     job_details.labels,
                     job_details.url,
                 )
-                _spawn_runner(
-                    runner_manager=runner_manager,
-                    job_url=job_details.url,
-                    msg=msg,
-                    github_client=github_client,
-                )
+                if not _validate_labels(
+                    labels=job_details.labels, supported_labels=supported_labels
+                ):
+                    logger.error(
+                        "Found unsupported job labels in %s. "
+                        "Will not spawn a runner and requeue the message.",
+                        job_details.labels,
+                    )
+                    msg.reject(requeue=True)
+                else:
+                    _spawn_runner(
+                        runner_manager=runner_manager,
+                        job_url=job_details.url,
+                        msg=msg,
+                        github_client=github_client,
+                    )
+
+
+def _validate_labels(labels: tuple[str, ...], supported_labels: tuple[str, ...]) -> bool:
+    """Validate the labels of the job.
+
+    Args:
+        labels: The labels of the job.
+        supported_labels: The supported labels for the runner.
+
+    Returns:
+        True if the labels are valid, False otherwise.
+    """
+    return all(label in supported_labels for label in labels)
```

Comment:
Perhaps make labels and supported labels a set and do a subset? That should be a much faster operation that this